src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -109,6 +115,22 @@ public override void Initialize(AnalysisContext context)
             });
         }
 
+        private static bool ThrowOperationOccursInDelegate(IThrowOperation throwOperation)
```

Comment:
Couple of suggestions:
- Should we consider using the below helper instead? https://github.com/dotnet/roslyn-analyzers/blob/5a5b8c68f488b4a8c7037c4b78008f1545e3344b/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L1121
- If not, consider using the below extension method instead of defining this new method: https://github.com/dotnet/roslyn-analyzers/blob/5a5b8c68f488b4a8c7037c4b78008f1545e3344b/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L245

Reply:
That's a good tip, thanks.